src/main/java/org/datavyu/plugins/nativeosx/NativeOSXMediaPlayer.java

```diff
@@ -139,11 +143,12 @@ protected float playerGetVolume() throws MediaException {
 
   @Override
   protected synchronized void playerSetVolume(float volume) throws MediaException {
+    logger.info("Setting Volume to " + volume);
     if (!muteEnabled) {
       if (volume == 0) {
```

Comment:
Simplify


if (volume == 0) {
        EventQueue.invokeLater(() -> nativePlayerCanvas.setVolume(0, id));
        EventQueue.invokeLater(() -> mediaPlayer.setVolume(0, id));
      } else {
        EventQueue.invokeLater(() -> nativePlayerCanvas.setVolume(volume * 10, id));
        EventQueue.invokeLater(() -> mediaPlayer.setVolume(volume , id));
        this.volume = mutedVolume = volume;
      }

into
EventQueue.invokeLater(() -> nativePlayerCanvas.setVolume(volume * 10, id));
EventQueue.invokeLater(() -> mediaPlayer.setVolume(volume*10 , id));
this.volume = mutedVolume = volume == 0 ? volume;

Reply:
One more comment? Why do we need to set the volume in two places?

Reply:
We need a local variable to keep track of the current volume since there is no native method to retrieve the volume.

Reply:
Done!

Reply:
We have two methods because we need to keep track of the local variable. That is not clear to me.

Reply:
There is only one method : mediaPlayer.setVolume, there is no reference to the nativePlayerCanvas since it was refactored to mediaPlayer.
Please take a look at the refactored method here: https://github.com/databrary/datavyu-ffmpegplugin/blob/963abcfdc8a21082051b6783651491234255aece/src/main/java/org/datavyu/plugins/nativeosx/NativeOSXMediaPlayer.java#L144